skbio/tree/_tree.py

```diff
@@ -1874,6 +1874,11 @@ def from_taxonomy(cls, lineage_map):
         root = cls(name=None)
         root._lookup = {}
 
+        if isinstance(lineage_map, dict):
+            lineage_map = lineage_map.items()
+        elif isinstance(lineage_map, pd.DataFrame):
+            lineage_map = ((idx, row.tolist()) for idx, row in lineage_map.iterrows())
```

Comment:
I'm hesitant about this as it assumes the frame is limited to lineage information, and that the columns are ordered properly. Enforcing column order though may require we limit users to specific column names though, which is viable also not great. If instead we require a column `taxonomy` or similar, and that it contains a delimited lineage then we could better control this e.g., `[(r.Index, delimiter.split(r.Taxonomy)) for r in lineage_map.itertuples()]`

Note that, minor right now, but `itertuples` is much faster than `iterrows`.

Reply:
@wasade Thanks for the review. I know that lineage strings are common, but taxonomic ranks separated by column is probably also common, and more friendly to downstream analysis. Not sure what use case @mortonjt encountered, pending his comment. It may be useful to have a function in skbio to convert lineage strings from/into columns.

Reply:
Hi yes I like the proposed approach of anticipating both taxonomic strings and data frame. Regarding the functionality I proposed earlier, I had taxonomic strings from metaphlan that I manually parsed into a data frame.  In my mind, having a dataframe as input could also be valuable, since it is easier to obtain summaries for each taxonomic level, as well as manipulate the taxonomic names. As @wasade mentioned earlier, the delimiters could widely vary -- I've seen both `;` and `|` used in practice.

Reply:
Will be something like:

```python
df['Taxonomy'].str.split('; ', expand=True).apply(lambda x: x.str.split('__').str[1])
```